### PR TITLE
switch to github actions

### PR DESCRIPTION
**Pull request by dbu**

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,57 +0,0 @@
...
-    - php: 7.3
-      env: SYMFONY_VERSION=4.4.*
-    - php: 7.3
-      env: SYMFONY_VERSION=4.3.*
-    - php: 7.2
-      env: SYMFONY_VERSION=4.2.*
-    - php: 7.1
-      env: SYMFONY_VERSION=3.4.*
-    - php: 7.1
-      env: SYMFONY_VERSION=2.8.*
```
Comment (dbu) [inline .travis.yml:33]: this can not have worked anymore, because the composer.json requires symfony components in version 3.4 or newer...

i will drop symfony 2 support in the test matrix.

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+    name: PHP ${{ matrix.php-version }} Test on Symfony ${{ matrix.symfony-version }} ${{ matrix.dependencies}} ${{ matrix.stability }} ${{ matrix.composer-flag }}
+    steps:
+      - name: Pull the code
+        uses: actions/checkout@v2
+      - name: Install PHP and Composer
+        uses: shivammathur/setup-php@v2
+        with:
+          php-version: ${{ matrix.php-version }}
+          tools: composer:v2, phpunit:${{ matrix.phpunit-version }}
+          ini-values: phar.readonly=off
```
Comment (dbu) [inline .github/workflows/ci.yml:58]: this setting needs to be off, or we can't create a phar. it would make sense if the phar task checks the php.ini setting and outputs a more helpful error when readonly if on. but that is out of scope for this PR.

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+        run: composer validate --strict --no-check-lock
+        if: ${{ matrix.stability != 'dev' }}
+      - name: Configure git
+        run: |
+          git config --global user.email "test@test.com"
+          git config --global user.name "John Doe"
+          git config --global init.defaultBranch master # TODO: use main, need to adjust tests to match
+
+      - name: Run tests
+        run: phpunit --coverage-text
```
Comment (dbu) [inline .github/workflows/ci.yml:82]: coverage actually only works with phpunit 9 because the phpunit.xml.dist file is in phpunit 9 format. on the other versions, it outputs a warning but does not fail. 

not sure if we should just drop the coverage-text flag maybe?

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+      - name: Composer update
+        run: composer update ${{ matrix.composer-flag }} --prefer-dist --no-interaction
+      - name: Composer validate
+        run: composer validate --strict --no-check-lock
+        if: ${{ matrix.stability != 'dev' }}
+      - name: Configure git
+        run: |
+          git config --global user.email "test@test.com"
+          git config --global user.name "John Doe"
+          git config --global init.defaultBranch master # TODO: use main, need to adjust tests to match
```
Comment (dbu) [inline .github/workflows/ci.yml:79]: some of the tests seem to assume that the default branch is always `master`. this fixes status quo, but we should change the tests to use main i think. out of scope for this pull request

File: .github/workflows/ci.yml [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+            symfony-version: '^6'
+          # Minimum supported dependencies with the oldest PHP version
+          - php-version: '7.1'
+            phpunit-version: 7
+            composer-flag: '--prefer-stable --prefer-lowest'
+          # Test latest unreleased versions
+          - php-version: '8.0'
+            phpunit-version: 9
+            symfony-version: '^6'
+            stability: 'dev'
```
Comment (jeanmonod) [inline .github/workflows/ci.yml:48]: This is a lot of versions. I just check on https://www.php.net/supported-versions.php, and I realized that versions below 7.4 are no more supported, so I don't see good reasons to still tests them. What about saving some CPU cycles and to support only the official versions: 7.4, 8.0 and 8.1
What do you think?

File: .github/workflows/ci.yml (hunk 4 above)
Comment (jeanmonod) [reply .github/workflows/ci.yml:79]: Yep, we can do this. I just create an issue for this: https://github.com/liip/RMT/issues/170

File: .github/workflows/ci.yml (hunk 2 above)
Comment (jeanmonod) [reply .github/workflows/ci.yml:58]: Yep, it's a big out of scope. But maybe putting a comment above this line to explain why we need to have this could make sens

File: .github/workflows/ci.yml (hunk 3 above)
Comment (jeanmonod) [reply .github/workflows/ci.yml:82]: Yes, let's drop it. Doesn't bring so much added value as we are no more under active dev on this project. Quite the opposite to be honest

File: README.md [hunk 6]
```diff
@@ -200,7 +200,7 @@ Actions can be used for pre or post release parts.
     * Option `commit-message`: specify a custom commit message. %version% will be replaced by the current / next version strings.
 * `vcs-tag`: Tag the last commit
 * `vcs-publish`: Publish the changes (commits and tags)
-* `composer-update`: Update the version number in a composer file
+* `composer-update`: Update the version number in a composer file (note that when using packagist.org, it is recommended to not have a tag in composer.json as the version is handle by version control tags)
```
Comment (jeanmonod) [inline README.md:203]: Thanks for updating the doc here 👍

**Review by jeanmonod — COMMENTED** [review]
@dbu I didn't spot anything wrong. Maybe the only thing to clarify is if we need to have so much config through the matrix. Personally, I would reduce the list, but you have a better experience than me on the topic...
Anyway, thanks a lot for the work

File: .github/workflows/ci.yml (hunk 5 above)
Comment (dbu) [reply .github/workflows/ci.yml:48]: i was thinking that this being an infrastructure tool, and not taking any extra effort to support old versions, we could keep supporting them so users can still upgrade RMT on legacy code bases.

but they can also keep installing the last released version if we drop support for old php versions. i let you decide.

the thing i find important is that composer.json matches with what we test - if we no longer test with < 7.4, we should also not have composer.json allow installation.

File: .github/workflows/ci.yml (hunk 2 above)
Comment (dbu) [reply .github/workflows/ci.yml:58]: adding a comment

File: .github/workflows/ci.yml (hunk 3 above)
Comment (dbu) [reply .github/workflows/ci.yml:82]: dropping it

File: .github/workflows/ci.yml (hunk 5 above)
Comment (jeanmonod) [reply .github/workflows/ci.yml:48]: You made a point here. And as it's already done. Let's keep it fort now

**Comment by jeanmonod** [comment]
@dbu Thanks a lot for all this work! You rock!
